include/secp256k1_ecdsa_sign_to_contract.h

```diff
@@ -0,0 +1,129 @@
+#ifndef SECP256K1_ECDSA_ANTI_COVERT_CHANNEL_H
+#define SECP256K1_ECDSA_ANTI_COVERT_CHANNEL_H
+
+#include "secp256k1.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Same as secp256k1_ecdsa_sign, but s2c_data32 is committed to by adding `hash(R1, s2c_data32)` to
+ *  the nonce generated by noncefp, with `ndata=hash(s2c_data32, ndata)`.
```

Comment:
Are we sure that there is no use case for `ndata` when you could also provide `s2c_data`? I am not familiar with uses of `ndata` in general, but if `s2c_data` alone is enough for all cases, then `ndata` could also be removed from `secp256k1_schnorrsig_sign`?

I can remove `noncefp`, but we could also keep it so other functions could be allowed in the future without breaking API compatibility.

Reply:
`ndata` is for giving additional information to the nonce function. Since it is hashed with `s2c_data` I don't see how you'd transmit usable information to the nonce function in that way. `ndata` shouldn't be removed from `schnorrsig_sign` because `schnorrsig_sign` can be used with other nonce functions than just the default. Imo, simpler is better than optimizing for some unknown future usecase.

Reply:
Dropped the `noncefp` and `ndata` params from the function in the alternative [here](https://github.com/bitcoin-core/secp256k1/pull/669). Can also do it for this version if we choose this alternative over the other.